### ISSUE TITLE
2019.2.0rc1 minion looping when trying to connect to localhost master 

### ISSUE THREAD

**Issue by vernondcole** [saltstack/salt#51186]
### Description of Issue/Question

Minion looping on connection attempt on self-mastered minion.

### Setup

MacOS 10.14.2 workstation # (any host OS will probably produce the same result)
Vagrant version 2.2.3
virtualbox version 5.2.22r126460
git version 2.17.2
salt 2018.3.3 # (salt-call will be used below in setting up the VM host)

### Steps to Reproduce Issue

Typing on your workstation:
```
git clone https://github.com/salt-bevy/salt-bevy.git
cd salt-bevy
git checkout develop

nano Vagrantfile
< about line 12, uncomment the reference which enables "git v2019.2.0rc1" >

./join_bevy.sh
```
* Take lots of defaults. The default will almost always be correct. . . .
* When the Python script asks `Will the Bevy Master be a VM guest of this machine?` answer `Y`es.
* Please select an IPv4 network adapter when prompted. (Your actual selection is unimportant.)

Continue as follows. . . 
```
vagrant up bevymaster
```
 ... watch the VM boot up, load Salt, and run a highstate ...
```
vagrant ssh bevymaster
```
Typing on the bevymaster VM:
```
sudo systemctl stop salt-minion
sudo salt-minion --log-level=debug
```
You should see the following during the looping of the running minion. . . .
```
[PROFILE ] Time (in seconds) to render '/vagrant/bevy_srv/pillar/manual_bevy_settings.sls' using 'yaml' renderer: 0.00293612480164
[CRITICAL] Unexpected error while connecting to localhost
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1018, in _connect_minion
    yield minion.connect_master(failed=failed)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1216, in connect_master
    yield self._post_master_init(master)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 1069, in run
    yielded = self.gen.send(value)
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1246, in _post_master_init
    async_pillar.destroy()
AttributeError: 'AsyncPillar' object has no attribute 'destroy'
[DEBUG   ] Connecting to master. Attempt 1 of 1
[DEBUG   ] Master URI: tcp://127.0.0.1:4506
[...]
```

### Versions Report
```
vernon@bevymaster:~$ salt --versions-report
Salt Version:
           Salt: 2019.2.0rc1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: 0.27.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-43-generic
         system: Linux
        version: Ubuntu 18.04 bionic
 
vernon@bevymaster:~$ 
```

**Comment by vernondcole** [comment]
On a subsequent test run on a new VM without editing the Vagrantfile (i.e. bootstrapping Salt 2018.3.3) , the bevymaster VM runs correctly.

**Comment by garethgreenaway** [comment]
@s0undt3ch Looks like this particular piece of code was added by you.  Any ideas?

**Comment by s0undt3ch** [comment]
Might have missed an implementation. Please assign to me.

**Comment by s0undt3ch** [comment]
The fix will be released as part of the up coming 2019.2.0
